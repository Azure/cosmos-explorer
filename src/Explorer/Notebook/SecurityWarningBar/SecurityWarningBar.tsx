import { MessageBar, MessageBarButton, MessageBarType } from "@fluentui/react";
import { actions, AppState } from "@nteract/core";
import React from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { NotebookUtil } from "../NotebookUtil";

export interface SecurityWarningBarPureProps {
  contentRef: string;
}

interface SecurityWarningBarDispatchProps {
  markNotebookAsTrusted: (contentRef: string) => void;
  saveNotebook: (contentRef: string) => void;
}

type SecurityWarningBarProps = SecurityWarningBarPureProps & StateProps & SecurityWarningBarDispatchProps;

interface SecurityWarningBarState {
  isBarDismissed: boolean;
}

export class SecurityWarningBar extends React.Component<SecurityWarningBarProps, SecurityWarningBarState> {
  constructor(props: SecurityWarningBarProps) {
    super(props);

    this.state = {
      isBarDismissed: false,
    };
  }

  render(): JSX.Element {
    return this.props.isNotebookUntrusted && !this.state.isBarDismissed ? (
      <MessageBar
        messageBarType={MessageBarType.warning}
        isMultiline={false}
        onDismiss={() => this.setState({ isBarDismissed: true })}
        actions={
          <div>
            <MessageBarButton
              onClick={() => {
                this.props.markNotebookAsTrusted(this.props.contentRef);
                this.props.saveNotebook(this.props.contentRef);
              }}
            >
              Trust Notebook
            </MessageBarButton>
          </div>
        }
      >
        <b>SECURITY WARNING</b>&nbsp; Be careful - this notebook is from an untrusted author. Run code cells at your own
        risk.
      </MessageBar>
    ) : (
      <></>
    );
  }
}

interface StateProps {
  isNotebookUntrusted: boolean;
}

interface InitialProps {
  contentRef: string;
}

// Redux
const makeMapStateToProps = (state: AppState, initialProps: InitialProps) => {
  const mapStateToProps = (state: AppState): StateProps => ({
    isNotebookUntrusted: NotebookUtil.isNotebookUntrusted(state, initialProps.contentRef),
  });
  return mapStateToProps;
};

const makeMapDispatchToProps = () => {
  const mapDispatchToProps = (dispatch: Dispatch): SecurityWarningBarDispatchProps => {
    return {
      markNotebookAsTrusted: (contentRef: string) => {
        return dispatch(
          actions.deleteMetadataField({
            contentRef,
            field: "untrusted",
          })
        );
      },
      saveNotebook: (contentRef: string) => dispatch(actions.save({ contentRef })),
    };
  };
  return mapDispatchToProps;
};

export default connect(makeMapStateToProps, makeMapDispatchToProps)(SecurityWarningBar);
