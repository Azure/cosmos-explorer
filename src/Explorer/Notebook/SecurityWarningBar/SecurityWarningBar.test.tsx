import { shallow } from "enzyme";
import React from "react";
import { SecurityWarningBar } from "./SecurityWarningBar";

describe("SecurityWarningBar", () => {
  it("renders if notebook is untrusted", () => {
    const wrapper = shallow(
      <SecurityWarningBar
        contentRef={"contentRef"}
        isNotebookUntrusted={true}
        markNotebookAsTrusted={undefined}
        saveNotebook={undefined}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders if notebook is trusted", () => {
    const wrapper = shallow(
      <SecurityWarningBar
        contentRef={"contentRef"}
        isNotebookUntrusted={false}
        markNotebookAsTrusted={undefined}
        saveNotebook={undefined}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
