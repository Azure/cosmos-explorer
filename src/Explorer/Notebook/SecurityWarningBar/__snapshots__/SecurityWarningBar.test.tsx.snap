// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityWarningBar renders if notebook is trusted 1`] = `<Fragment />`;

exports[`SecurityWarningBar renders if notebook is untrusted 1`] = `
<StyledMessageBar
  actions={
    <CustomizedMessageBarButton
      onClick={[Function]}
    >
      Trust Notebook
    </CustomizedMessageBarButton>
  }
  dismissButtonAriaLabel="Close"
  isMultiline={false}
  messageBarType={5}
  onDismiss={[Function]}
>
   
  This notebook was downloaded from the public gallery. Running code cells from a notebook authored by someone else may involve security risks.
</StyledMessageBar>
`;
